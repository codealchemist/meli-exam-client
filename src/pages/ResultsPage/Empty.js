import React from 'react'
import styled from 'styled-components'
import noResults from './empty-box.gif'

const Wrapper = styled.div`
  width: 100%;
  max-width: 1200px;
  text-align: center;
  padding: 50px 0;
  margin: auto;

  h1 {
    margin: 0;
    font-weight: lighter;
  }
`

const Empty = ({ title }) => (
  <Wrapper>
    <img src={noResults} alt="No results" />
    <h1>{title}</h1>
  </Wrapper>
)

export default Empty
