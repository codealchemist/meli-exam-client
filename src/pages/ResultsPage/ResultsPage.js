import React from 'react'
import styled from 'styled-components'
import Results from '../../components/Results'
import Context from '../../Context'
import Colors from '../../components/Colors'
import Breadcrumb from '../../components/Breadcrumb'
import Empty from './Empty'
import { search as searchApi } from '../../api'

const Wrapper = styled.div`
  width: 100%;
  background: ${Colors.default.light2};
  padding-bottom: 40px;
`

class ResultsPage extends React.PureComponent {
  state = {
    searching: false
  }

  componentDidMount() {
    if (this.context.searchResponse) return

    const { location } = this.props
    const params = new URLSearchParams(location.search)
    const query = params.get('search')
    this.search(query)
  }

  search = query => {
    this.setState({ searching: true })
    searchApi(query).then(response => {
      console.log('SEARCH RESPONSE', response)
      this.context.setSearchResponse(response)
      this.setState({ searching: false })
    })
  }

  render() {
    return (
      <>
        {this.context.searchResponse && (
          <Wrapper>
            {!!this.context.searchResponse.filters.length && (
              <Breadcrumb
                categories={
                  this.context.searchResponse.filters[0].values[0]
                    .path_from_root
                }
              />
            )}
            <Results items={this.context.searchResponse.results || []} />
          </Wrapper>
        )}

        {!this.context.searchResponse && !this.state.searching && (
          <Empty title="SchrÃ¶dinger's Cat is gone. Sorry." />
        )}
      </>
    )
  }
}

ResultsPage.contextType = Context
export default ResultsPage
