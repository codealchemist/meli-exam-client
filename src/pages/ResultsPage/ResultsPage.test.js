import React from 'react'
import { mount } from 'enzyme'
import ResultsPage from './ResultsPage'

jest.mock('../../Context', () => ({
  Consumer: obj =>
    obj.children({
      setSearchResponse: jest.fn(),
      searchResponse: {
        items: []
      }
    })
}))

const context = {
  setSearchResponse: jest.fn(),
  searchResponse: {
    items: [],
    filters: []
  }
}

ResultsPage.contextType = React.createContext(context)

describe('ResultsPage', () => {
  it('should render', () => {
    const wrapper = mount(<ResultsPage />)
    expect(wrapper).toMatchSnapshot()
  })
})
