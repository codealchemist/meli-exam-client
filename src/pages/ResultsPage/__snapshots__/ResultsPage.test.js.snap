// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage should render 1`] = `
<ResultsPage>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-dnqmqq",
            "isStatic": true,
            "lastClassName": "eWHIHI",
            "rules": Array [
              "
  width: 100%;
  background: ",
              "#eee",
              ";
  padding-bottom: 40px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-dnqmqq",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-dnqmqq eWHIHI"
      >
        <Results
          items={Array []}
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-ifAKCX",
                    "isStatic": true,
                    "lastClassName": "dBJSQK",
                    "rules": Array [
                      "
  max-width: 1200px;
  width: 100%;
  margin: auto;
  background: ",
                      "#fff",
                      ";
  color: ",
                      "#333",
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-ifAKCX",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-ifAKCX dBJSQK"
              />
            </StyledComponent>
          </styled.div>
        </Results>
      </div>
    </StyledComponent>
  </styled.div>
</ResultsPage>
`;
